esp-hal/src/spi/master.rs

```diff
@@ -2346,101 +2335,83 @@ pub trait InstanceDma: Instance {
         }
     }
 
-    #[cfg(gdma)]
     fn enable_dma(&self) {
-        let reg_block = self.register_block();
-        reg_block.dma_conf().modify(|_, w| w.dma_tx_ena().set_bit());
-        reg_block.dma_conf().modify(|_, w| w.dma_rx_ena().set_bit());
+        #[cfg(gdma)]
+        {
+            // for non GDMA this is done in `assign_tx_device` / `assign_rx_device`
+            let reg_block = self.register_block();
+            reg_block.dma_conf().modify(|_, w| {
+                w.dma_tx_ena().set_bit();
+                w.dma_rx_ena().set_bit()
+            });
+        }
+        #[cfg(pdma)]
+        self.reset_dma();
     }
 
-    #[cfg(pdma)]
-    fn enable_dma(&self) {
-        // for non GDMA this is done in `assign_tx_device` / `assign_rx_device`
+    fn reset_dma(&self) {
+        #[cfg(pdma)]
+        {
+            let reg_block = self.register_block();
+            reg_block.dma_conf().modify(|_, w| {
+                w.out_rst().set_bit();
+                w.in_rst().set_bit();
+                w.ahbm_fifo_rst().set_bit();
+                w.ahbm_rst().set_bit()
+            });
+            reg_block.dma_conf().modify(|_, w| {
+                w.out_rst().clear_bit();
+                w.in_rst().clear_bit();
+                w.ahbm_fifo_rst().clear_bit();
+                w.ahbm_rst().clear_bit()
+            });
+        }
+        #[cfg(gdma)]
+        {
+            let reg_block = self.register_block();
+            reg_block.dma_conf().modify(|_, w| {
+                w.rx_afifo_rst().set_bit();
+                w.buf_afifo_rst().set_bit();
+                w.dma_afifo_rst().set_bit()
+            });
+            reg_block.dma_conf().modify(|_, w| {
+                w.rx_afifo_rst().clear_bit();
+                w.buf_afifo_rst().clear_bit();
+                w.dma_afifo_rst().clear_bit()
+            });
+        }
+        self.clear_dma_interrupts();
     }
 
     #[cfg(gdma)]
     fn clear_dma_interrupts(&self) {
         let reg_block = self.register_block();
         reg_block.dma_int_clr().write(|w| {
-            w.dma_infifo_full_err()
-                .clear_bit_by_one()
-                .dma_outfifo_empty_err()
-                .clear_bit_by_one()
-                .trans_done()
-                .clear_bit_by_one()
-                .mst_rx_afifo_wfull_err()
-                .clear_bit_by_one()
-                .mst_tx_afifo_rempty_err()
-                .clear_bit_by_one()
+            w.dma_infifo_full_err().clear_bit_by_one();
```

Comment:
Woah! Did you manage to convince bjoernQ that this style is better/clearer?

Reply:
no, I'm strongarming this :D

Reply:
@bjoernQ did you have something against this style? I find pairing the value with the bitfield in the same line a lot easier to decode at a quick glance.

Reply:
I was never really advocating for any style but I agree this is more readable - I also think we had an issue regarding the style somewhere. 

Ok - can't find it for some reason

I was never really advocating for any style but I agree this is more readable - I also think we had an issue regarding the style somewhere. I think the original style "just happened"

Ideally, we should just not mix different styles in the same file (but well - blowing up a five-line PR by changing style in thousands of lines is also problematic)

Reply:
I plan on cleaning this up as I go. I remember @jessebraham removing the `#[rustfmt::skip]` that then reformatted everything into this linear thing. Jesse, do you have a preference either way?

Reply:
Doesn't matter to me